File: tech_exercise/package/exercise1/api/Business/Dtos/PersonAstronaut.cs
namespace StargateAPI.Business.Dtos
{
    public class PersonAstronaut
    {
        public int PersonId { get; set; }

        public string Name { get; set; } = string.Empty;

        public string CurrentRank { get; set; } = string.Empty;

        public string CurrentDutyTitle { get; set; } = string.Empty;

        public DateOnly? CareerStartDate { get; set; } //changed DateTime to DateOnly for simplicity.

        public DateOnly? CareerEndDate { get; set; } //changed DateTime to DateOnly for simplicity.

        public bool IsRetired => string.Equals(CurrentDutyTitle, "RETIRED", StringComparison.OrdinalIgnoreCase);
    }
}

File: tech_exercise/package/exercise1/ui/src/app/core/models/person-astronaut.ts
export interface PersonAstronaut {
  personId: number;
  name: string;
  currentRank: string;
  currentDutyTitle: string;
  careerStartDate: string | null;
  careerEndDate: string | null;
  isRetired: boolean;
}

export const isAstronaut = (person: PersonAstronaut): boolean =>
  !!person.currentRank && person.currentRank.trim().length > 0;
